pkg/util/log/log.go

```diff
@@ -44,6 +44,15 @@ type DatadogLogger struct {
 // SetupLogger setup agent wide logger
 func SetupLogger(i seelog.LoggerInterface, level string) {
 	logger = setupCommonLogger(i, level)
+
+	// Flush the log entries logged before initialization now that the logger is initialized
+	bufferMutex.Lock()
+	bufferLogsBeforeInit = false
+	defer bufferMutex.Unlock()
+	for _, logLine := range logsBuffer {
+		logLine()
+	}
+	logsBuffer = []func(){}
```

Comment:
Why this code is not needed for `SetupJMXLogger`?

Reply:
This logic allows actually logging the log messages that are submitted to the main agent logger before it's initialized (for example, log messages logged when the main config is initialized). This is irrelevant to the jmxfetch logger (the jmxfetch logger is not the default one, and is only used after it's initialized)